src/mds/MDBalancer.cc

```diff
@@ -663,7 +664,8 @@ void MDBalancer::prep_rebalance(int beat)
 	dout(15) << "   mds." << it->second << " is importer" << dendl;
 	importers.insert(pair<double,mds_rank_t>(it->first,it->second));
 	importer_set.insert(it->second);
-      } else {
+      } else if ((it->first > target_load * (1.0 + g_conf->mds_bal_min_rebalance)) && 
+        (it->second == whoami || !mds_last_epoch_under_info[it->second] || beat_epoch - mds_last_epoch_under_info[it->second] >= 2)){
```

Comment:
It's better to assign 'mds_last_epoch_under_info[it->second]' to a local variable。 such as

'int mds_last_epoch_under = (it->second != whoami) ? 0 : mds_last_epoch_under_info[it->second];'

Reply:
adjust the judgement logic as you suggested, could you please take a look again?